String interpolation doesn't allow const char*

```c
fn main() {
    let text: const char* = "example.com";
    let transformed_text = "https://{text}";

    println "{transformed_text}";
}
```

the code above returns an error:

```
out.c: In function 'main':
out.c:39:28: error: '_Generic' selector of type 'const char *' is not compatible with any association
   39 | #define _z_str(x) _Generic((x), _Bool: "%s", char: "%c", signed char: "%c", unsigned char: "%u", short: "%d", unsigned short: "%u", int: "%d", unsigned int: "%u", long: "%ld", unsigned long: "%lu", long long: "%lld", unsigned long long: "%llu", float: "%f", double: "%f", char*: "%s", void*: "%p" _z_objc_map)
      |                            ^
out.c:97:13: note: in expansion of macro '_z_str'
   97 | sprintf(_t, _z_str(text), text);
      |             ^~~~~~
C compilation failed.
```

However, if we were to remove the `const` from the `text` variable, then it works fine.

I was wondering if there's any reason why zen c doesn't allow string interpolation with `const char*`.

--- a/src/codegen/codegen_decl.c
+++ b/src/codegen/codegen_decl.c
@@ -30,7 +30,7 @@ static void emit_freestanding_preamble(FILE *out)
           "unsigned short: \"%u\", int: \"%d\", unsigned int: \"%u\", "
           "long: \"%ld\", unsigned long: \"%lu\", long long: \"%lld\", "
           "unsigned long long: \"%llu\", float: \"%f\", double: \"%f\", "
-          "char*: \"%s\", void*: \"%p\")\n",
+          "char*: \"%s\", const char*: \"%s\", void*: \"%p\")\n",
           out);
     fputs("#define _z_arg(x) _Generic((x), _Bool: _z_bool_str(x), default: (x))\n", out);
     fputs("typedef struct { void *func; void *ctx; } z_closure_T;\n", out);

--- a/src/codegen/compat.h
+++ b/src/codegen/compat.h
@@ -73,7 +73,7 @@
     "unsigned short: \"%u\", int: \"%d\", unsigned int: \"%u\", "                                  \
     "long: \"%ld\", unsigned long: \"%lu\", long long: \"%lld\", "                                 \
     "unsigned long long: \"%llu\", float: \"%f\", double: \"%f\", "                                \
-    "char*: \"%s\", void*: \"%p\" _z_objc_map)\n"
+    "char*: \"%s\", const char*: \"%s\", void*: \"%p\" _z_objc_map)\n"
 
 #define ZC_C_ARG_GENERIC_STR                                                                       \
     "#define _z_arg(x) _Generic((x), _Bool: _z_bool_str(x) _z_objc_arg_map(x), default: (x))\n"

--- a/tests/language/features/test_implicit_fstring.zc
+++ b/tests/language/features/test_implicit_fstring.zc
@@ -6,6 +6,12 @@ test "implicit_fstring_interpolation" {
     assert(strcmp(s, "123") == 0, "Implicit f-string failed");
 }
 
+test "with_const_char_ptr" {
+    let a: const char* = "test";
+    let s = "Text: {a}";
+    assert(strcmp(s, "Text: test") == 0, "With const char* f-string failed");
+}
+
 test "implicit_fstring_complex" {
     let a = 10;
     let b = 20;

